agent/internal/client/interface.go

```diff
@@ -78,6 +79,9 @@ type SharedCommunicator interface {
 	// GetDataPipesConfig returns the Data-Pipes service configuration.
 	GetDataPipesConfig(context.Context) (*apimodels.DataPipesConfig, error)
 
+	// GetPullRequest takes in a PR number, owner, and repo and returns the corresponding pull request.
+	GetPullRequest(context.Context, TaskData, int, string, string) (*github.PullRequest, error)
```

Comment:
It seems like we could get away with adding an api model that just has a mergeable boolean and a ref.
If so, it would seem advantageous: our API won't be dependent on the GitHub SDK. So like, if they change the PR format (admittedly unlikely) we won't need to worry about the agent and app servers being on the same version.

Reply:
Yeah i was torn about which one to do, since if we want more info from the PR we'll have to add to our defined structure. I do like the idea of not having to send as much though.